statistics/cmsketch.go

```diff
@@ -99,18 +96,19 @@ func newTopNHelper(sample [][]byte, numTop uint32) *topNHelper {
 	// frequency of the n-th element are added to the TopN statistics. We chose
 	// 2/3 as an empirical value because the average cardinality estimation
 	// error is relatively small compared with 1/2.
-	for i := uint32(0); i < sampleNDV && i < numTop*2; i++ {
-		if i >= numTop && sorted[i]*3 < sorted[numTop-1]*2 && last != sorted[i] {
+	var actualNumTop uint32
+	for ; actualNumTop < sampleNDV && actualNumTop < numTop*2; actualNumTop++ {
+		if actualNumTop >= numTop && sorted[actualNumTop].cnt*3 < sorted[numTop-1].cnt*2 && last != sorted[actualNumTop].cnt {
 			break
 		}
-		if sorted[i] == 1 {
+		if sorted[actualNumTop].cnt == 1 {
 			break
 		}
-		last = sorted[i]
-		sumTopN += sorted[i]
+		last = sorted[actualNumTop].cnt
```

Comment:
Now we do not need the `last`? Because we have the actual number of topn?

Reply:
Yes.

Reply:
So can we remove it here?